superset-frontend/src/datasource/DatasourceModal.tsx

```diff
@@ -64,17 +64,17 @@ interface DatasourceModalProps {
   show: boolean;
 }
 
-function buildMetricExtraJsonObject(metric: Record<string, unknown>) {
+function buildExtraJsonObjects(item: Record<string, unknown>) {
```

Comment:
cool refactor, maybe name `buildExtraJsonObject` though since it only returns 1 object?